src/components/AddClass/AddClassTemplate.tsx

```diff
@@ -0,0 +1,88 @@
+import { Input } from "@/components/ui/input";
+import React, {
+  useRef,
+  useEffect,
+  useState,
+  Dispatch,
+  SetStateAction,
+} from "react";
+import { AddClassTemplateProps } from "@/app/utils/interfaces";
+
+/*
+ * This component is the "popup" that appears when a user clicks the "Add Class" button.
+ * The goal is to have the user fill out the number of units, course name, and course ID.
+ */
+
+const AddClassTemplate: React.FC<AddClassTemplateProps> = ({ onSubmit }) => {
+  const inputRefs = [
+    useRef<HTMLInputElement>(null),
+    useRef<HTMLInputElement>(null),
+    useRef<HTMLInputElement>(null),
+  ];
+
+  const [inputError, setInputError]: [
+    boolean,
+    Dispatch<SetStateAction<boolean>>
+  ] = useState(false);
+
+  // On component render, this will focus on the first input box
+  useEffect(() => {
+    if (inputRefs[0].current) {
+      inputRefs[0].current.focus();
+    }
+  }, []);
+
+  // This function allows for arrow keys to be used to go up and down input boxes
+  const handleKeyDown = (
+    e: React.KeyboardEvent<HTMLInputElement>,
+    index: number
+  ) => {
+    if (e.key === "ArrowDown" && index < inputRefs.length - 1)
+      inputRefs[index + 1]?.current?.focus(); // Move to the next input
+    if (e.key === "ArrowUp" && index > 0)
+      inputRefs[index - 1]?.current?.focus(); // Move to the previous input
+    // TODO: Add a "Submit" button or allow the user to click outside the popup to add classes. Right now, the "Enter" key works.
```

Comment:
I think this should be a separate from our MVP, what do we all think?

Reply:
What specifically do you think should be separate? The navigation via keys? If so, yeah, I do not think that is essential.

Reply:
I meant that TODO snippet, right now I have it set to pressing "Enter" to create a class. The separate target could be to add a submit button, or the ability to click "off" the popup to add the class. The navigation via keys is already implemented!